pkg/controllers/hpaautolabelretain/hpa_auto_label_retain_predicate.go

```diff
@@ -44,7 +44,7 @@ func (r *HPAReplicasSyncer) Create(e event.CreateEvent) bool {
 }
 
 // Update implements UpdateEvent filter
-func (r *HPAReplicasSyncer) Update(e event.UpdateEvent) bool {
+func (r *HPAAutoLabelRetain) Update(e event.UpdateEvent) bool {
```

Comment:
How to directly filter out ScaleTargetRef as non-Deployment?

Reply:
Due to we have document for users [Use the Retain Interpreter to Resolve Control Conflict between the Control Plane and Member Clusters](https://karmada.io/docs/userguide/globalview/customizing-resource-interpreter/#use-the-retain-interpreter-to-resolve-control-conflict-between-the-control-plane-and-member-clusters) to teach how to configure retain, may be we shall keep this logic.